### ISSUE TITLE
Grouping behaviour is hard to discover and understand

### ISSUE THREAD

**Issue by tomwilkie** [weaveworks/scope#156]
This is (my paraphrasing of) feedback from @rade, @squaremo and @awh.

Comments generally include "thats really cool" but "I didn't know to click that small button to make that happen" and "it took a minute for me to figure out what it was doing".

**Comment by rade** [comment]
Actually, without you explaining what it was doing I would never have figured it out.

**Comment by davkal** [comment]
Should we just concede defeat and bring the labels back?

**Comment by peterbourgon** [comment]
```
Applications   Containers    Hosts  
by name        by image          
```

**Comment by davkal** [comment]
Current topos collection object is:

```json
{
    "name": "Applications",
    "url": "/api/topology/applications",
    "grouped_url": "/api/topology/applications-grouped",
    "stats": {
      "node_count": 12,
      "nonpseudo_node_count": 10,
      "edge_count": 13
    }
  }
```

I propose to nest sub-topologies:

```json
{
    "name": "Applications",
    "url": "/api/topology/applications",
    "stats": {
      "node_count": 12,
      "nonpseudo_node_count": 10,
      "edge_count": 13
    }
    "sub_topologies": [
       {
           "name": "by name"
           "url": "/api/topology/applications-by-name"
       }
   ]
}
```

**Comment by davkal** [comment]
FYI, the UI logic uses the last part of the url as the topology ID, e.g. `id(/api/topology/applications-by-name) -> applications-by-name` so it would be helpful if we can keep this prefix structure.

**Comment by davkal** [comment]
Fixed via #222
